### PR TITLE
[[ Win64 ]] Initial changes for Win32 x86-64

### PR DESCRIPTION
**Pull request by runrevmark**
This patch contains initial changes for Win32 x86-64 builds.

Depends on: https://github.com/runrevmark/livecode/pull/10

### CODE REVIEW

File: config/win32.gypi [hunk 1]
```diff
@@ -25,6 +25,18 @@
...
+		'include_dirs':
+		[
+			'$(foo)C:/WinDDK/7600.16385.1/inc/atl71',
+			'$(foo)C:/WinDDK/7600.16385.1/inc/mfc42',
+		],
+		
+		'library_dirs':
+		[
+			'$(foo)C:/WinDDK/7600.16385.1/lib/ATL/i386',
+			'$(foo)C:/WinDDK/7600.16385.1/lib/MFC/i386',
```
Comment (peter-b) [inline config/win32.gypi]: Should these be conditional on Windows architecture name?

File: configure.bat [hunk 2]
```diff
@@ -22,6 +22,13 @@ IF %TARGET_ARCH%==x86_64 (
   SET MSVC_ARCH=%TARGET_ARCH%
 )
 
+REM The LiveCode name for x86_64 is x86_64
+IF %MSVC_ARCH%==x64 (
+    SET LC_ARCH=x86_64
+) ELSE (
+    SET LC_ARCH=x86
+)
+
```
Comment (peter-b) [inline configure.bat]: Why is `%LC_ARCH%` needed? I think the intention with this script was that `%TARGET_ARCH%` should be the LiveCode architecture name, given the way that it is used to compute `%MSVC_ARCH%` (please correct me if that is incorrect!)

File: engine/src/combiners.cpp [hunk 3]
```diff
@@ -301,32 +301,9 @@ static INLINE uint16_t long_scaled_divide(uint32_t n, uint8_t s, uint16_t d)
...
-  less_than_9_bits:
-		xor ax, ax
-		mov  ax, g_sqrt_table[ebx*2]
-		mov cx, dx
-		shr ax, cl
-	}
-#else
 	return 1;
-#endif
 }
```
Comment (peter-b) [inline engine/src/combiners.cpp:307]: Maybe this function should be removed entirely; it no longer does anything useful on any platform.

File: libcore/src/core.cpp [hunk 4]
```diff
@@ -66,6 +66,7 @@ bool MCThrow(uint32_t p_error)
 #include <crtdbg.h>
 #include <dbghelp.h>
 
+// WIN64-CHECK
```
Comment (peter-b) [inline libcore/src/core.cpp]: Does this comment indicate that some additional work is required in the future?  If so, it should probably contain `TODO`.

File: libgraphics/src/legacyblendmodes.cpp [hunk 5]
```diff
@@ -375,29 +375,7 @@ static INLINE uint16_t long_scaled_divide(uint32_t n, uint8_t s, uint16_t d)
...
-	less_than_9_bits:
-		xor ax, ax
-		mov  ax, s_sqrt_table[ebx*2]
-		mov cx, dx
-		shr ax, cl
-	}
-#else
 	return 1;
-#endif
 }
```
Comment (peter-b) [inline libgraphics/src/legacyblendmodes.cpp:379]: As previously, I think this function would be best removed entirely.

File: config/win32.gypi (hunk 1 above)
Comment (runrevmark) [reply config/win32.gypi]: I don't think those are necessary - so I'm trying removing them. The default setup of Vs2015/2017 seems to work fine without them.

File: engine/src/combiners.cpp (hunk 3 above)
Comment (runrevmark) [reply engine/src/combiners.cpp:307]: Possibly - I suspect this file and legacyblendmodes.cpp are going to be obsoleted by the Skia changes... So, I think dealing with it there is a better idea.

File: libcore/src/core.cpp (hunk 4 above)
Comment (runrevmark) [reply libcore/src/core.cpp]: They were just markers for things I needed to double-check - will do so and remove :)

File: libgraphics/src/legacyblendmodes.cpp (hunk 5 above)
Comment (runrevmark) [reply libgraphics/src/legacyblendmodes.cpp:379]: See above

File: configure.bat (hunk 2 above)
Comment (runrevmark) [reply configure.bat]: There are still things which rely on TARGET_ARCH being the (native) arch name. This needs a bit of a moomin - the current configure.bat ensures that MSVC_ARCH and TARGET_ARCH are both set to X64 for 64-bit, where as LC_ARCH uses X86_64.

**Pull request merged**